src/routes/index.svelte

```diff
@@ -134,166 +21,528 @@
+                    acc.departments[department].positions.push(position);
+                }
+
+                return acc;
+            },
+            { experiences: [], departments: {} }
+        );
+
+        return { EXPERIENCE: experiences.sort(), DEPARTMENTS: departments };
+    }
+
     function filterTalents({
-        location,
-        role,
-        isOpenToRelocation,
-        isOpenToRemoteWork
+        position,
+        department,
+        experience,
+        isOpenToRelocationOrRemote
     }) {
         filteredTalents = talents.filter(talent => {
-            let isLocationValid = false;
-            let isRoleValid = false;
-
-            let isReloc = true;
-            let isRemote = true;
+            let isPositionValid = false;
+            let isDepartmentValid = false;
+            let isXPValid = false;
+            let isOpenToRelocationOrRemoteValid = true;
 
-            if (location === "all" || location === talent.location.key) {
-                isLocationValid = true;
+            if (
+                position === allPositionsOption ||
+                position === talent.position
+            ) {
+                isPositionValid = true;
             }
-            if (role === "all" || role === talent.role.key) {
-                isRoleValid = true;
+            if (
+                department === allDepartmentsOption ||
+                department === talent.department
+            ) {
+                isDepartmentValid = true;
             }
-            if (isOpenToRelocation && !talent.isOpenToRelocation) {
-                isReloc = false;
+            if (
+                experience === allExperiencesOption ||
+                experience === talent.experience
+            ) {
+                isXPValid = true;
             }
-            if (isOpenToRemoteWork && !talent.isOpenToRemoteWork) {
-                isRemote = false;
+            if (
+                isOpenToRelocationOrRemote &&
+                !talent.isOpenToRelocationOrRemote
+            ) {
+                isOpenToRelocationOrRemoteValid = false;
             }
 
-            return isLocationValid && isRoleValid && isReloc && isRemote;
+            return (
+                isPositionValid &&
+                isDepartmentValid &&
+                isXPValid &&
+                isOpenToRelocationOrRemoteValid
+            );
         });
     }
 
-    let location = "all";
-    let role = "all";
-    let isOpenToRelocation = false;
-    let isOpenToRemoteWork = false;
+    function handleWindowScroll() {
+        isFilterBarActive =
+            filterBar && filterBar.getBoundingClientRect().top <= 0;
+    }
 
-    $: filterTalents({
-        location,
-        role,
-        isOpenToRelocation,
-        isOpenToRemoteWork
-    });
+    function handleResetPosition() {
+        positionSelectValue = allPositionsOption;
+    }
 </script>
 
+<svelte:window
+    on:scroll="{handleWindowScroll}"
```

Comment:
this will be called *a lot* when scrolling. You should either debounce it, or have some fun with https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

Reply:
I could have used it but on desktop the filter-bar is sticky. So we can't scroll over and the IO API is never triggered.
If you have any tips to share about IO API's options that could fix up that, feel free 😄